merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/remotes/postgres/InsertConstraintRunsAction.java

```diff
@@ -32,7 +34,7 @@ public void apply(
       statement.setString(2, constraint.definition());
       statement.setLong(3, simulationDatasetId);
 
-      if (results.get(constraint.id()) != null) {
+      if (results.get(constraint.id()) != null && !results.get(constraint.id()).isEmpty()) {
```

Comment:
I retain empty results to indicate a successful compilation of the constraint without any violations. At this point, I store constraints with violations in the database; otherwise, I store {}.

Reply:
Is this change why the E2E tests now contain a `results` object in the successful cases?

Reply:
Actually, thinking about it, a constraint run with no violations should have the exact same data structure as a constraint run with violations, however, the `violations` key should be an empty array.

Reply:
Yes that is correct. There are 3 object that can be returned in the response array. This change affected the e2e test:

compiled but no violations
```js
{
success: true
result : {}
errors : []
}
```

compiled with violations
```js
{
success: true
result : {
   violations: [...],
   constraintID: 1,
   constriantName: "",
   gaps: [...]
}
errors : []
}
```

failed compiled
```js
{
success: false,
result : {},
errors : [
  {
        message: "",
        location: {}
        stack: ""
  }
]
}
```

Reply:
Added an e2e test for this bug.

Reply:
Forgot to respond to this, but compiled with no violations or gaps should look like:

```json
{
"success": true
"result" : {
  "violations:" [],
  "constraintID": 1,
  "constriantName": "",
  "gaps": []
}
"errors": []
}
```

This keep's the object's form consistent in the `success: true` case and does not cause information not related to violations or gaps to be lost.

Reply:
Got it, I changed the code so it won't drop this information and send it back to the ui. 

https://github.com/NASA-AMMOS/aerie/pull/1242/files#diff-e4402938fdcefa189f062abfca8a610f16d1dd9272ccdaeb0c4b6ca59b383d4bL220